agr_literature_service/lit_processing/dump_json_data.py

```diff
@@ -0,0 +1,576 @@
+                                             reference_id_to_xrefs,
+                                             reference_id_to_authors,
+                                             reference_id_to_comment_correction_data,
+                                             reference_id_to_mod_reference_types,
+                                             reference_id_to_mesh_terms,
+                                             reference_id_to_mod_corpus_data,
+                                             resource_id_to_journal,
+                                             json_path + json_file, datestamp)
+    except Exception as e:
+        print("Error occurred when retrieving data from Reference data and generating json file: " + str(e))
+        return
+
+    log.info("Uploading json file to s3...")
+    try:
+        upload_json_file_to_s3(json_path, json_file, datestamp, ondemand)
+    except Exception as e:
+        print("Error occurred when uploading json file to s3: " + str(e))
+        return
+
+    log.info("DONE!")
+
+
+def generate_json_file(metaData, data, filename_with_path):
+
+    try:
+        jsonData = {"data": data,
+                    "metaData": metaData}
+        fw = open(filename_with_path, 'w')
+        fw.write(json.dumps(jsonData, indent=4, sort_keys=True))
+        fw.close()
+    except Exception as e:
+        print("Error when generating", filename_with_path, ": " + str(e))
+        exit()
+
+
+def upload_json_file_to_s3(json_path, json_file, datestamp, ondemand):
+
+    env_state = environ.get('ENV_STATE', 'develop')
+    if env_state == 'build':
+        env_state = 'develop'
+    if env_state == 'test':
+        return
+
+    json_file_with_datestamp = json_file.replace('.json', '_' + datestamp + '.json')
+
+    if ondemand:
+        s3_filename = ondemand_bucket.replace('develop', env_state) + json_file_with_datestamp
+        upload_file_to_s3(json_path + json_file, s3_bucket, s3_filename)
+        return
+
+    ## upload file to recent bucket
+    s3_filename = recent_bucket.replace('develop', env_state) + json_file_with_datestamp
+    upload_file_to_s3(json_path + json_file, s3_bucket, s3_filename)
+
+    ## upload file to latest bucket
+    s3_filename = latest_bucket.replace('develop', env_state) + json_file
+    upload_file_to_s3(json_path + json_file, s3_bucket, s3_filename)
+
+    ## upload file to monthly bucket if it is first day of the month
+    todayDate = date.today()
+    if todayDate.day == 1:
+        s3_filename = monthly_bucket.replace('develop', env_state) + json_file_with_datestamp
+        upload_file_to_s3(json_path + json_file, s3_bucket, s3_filename, 'GLACIER_IR')
+
+
+def concatenate_json_files(json_file, index):
+
+    if index == 1:
+        rename(json_file + '_0', json_file)
+        return
+
+    fw = open(json_file, "w")
+
+    for i in range(index):
+        this_json_file = json_file + '_' + str(i)
+        f = open(this_json_file)
+        if i == 0:
```

Comment:
Nice to see how you wrote this !  When I suggested, I didn't know how to do it efficiently.